wgpu/src/image.rs

```diff
@@ -20,6 +23,9 @@ pub struct Pipeline {
     instances: wgpu::Buffer,
     constants: wgpu::BindGroup,
     texture_layout: wgpu::BindGroupLayout,
+    #[debug_stub="ReplacementValue"]
+    allocator: AtlasAllocator,
+    atlas: wgpu::Texture,
```

Comment:
I think both of these fields are quite coupled. It may be a good idea to couple them in a new opaque type: `image::Atlas`. This way we should be able to narrow the scope and control mutation nicely.